app/src/main/java/org/onionshare/android/ui/MainActivity.kt

```diff
@@ -29,22 +35,43 @@ class MainActivity : ComponentActivity() {
                         onFabClicked = this::onFabClicked,
                         onFileRemove = viewModel::removeFile,
                         onRemoveAll = viewModel::removeAll,
-                        onSheetButtonClicked = viewModel::onSheetButtonClicked,
+                        onSheetButtonClicked = this::onSheetButtonClicked,
                     )
                 }
             }
         }
     }
 
-    private val launcher = registerForActivityResult(GetMultipleContents()) { uris ->
+    private val contentLauncher = registerForActivityResult(GetMultipleContents()) { uris ->
         viewModel.onUrisReceived(uris)
     }
 
+    private val batteryLauncher = registerForActivityResult(StartActivityForResult()) {
+        // we just ignore the result and don't check for battery optimization again
+        // assuming the user will understand if they didn't allow background
```

Comment:
I think this might be optimistic, but maybe we should wait for user testing before putting more work into it?

Reply:
I added a TODO here as well